lib/ruby_lsp/global_state.rb

```diff
@@ -21,7 +21,7 @@ class GlobalState
     attr_reader :encoding
 
     sig { returns(T::Boolean) }
-    attr_reader :supports_watching_files, :experimental_features
+    attr_reader :supports_watching_files, :experimental_features, :supports_request_delegation
```

Comment:
Since this is custom and not a part of the specification, we need to know if the client has support for it before raising the custom signals or sending custom notifications.